### PR TITLE
[FLINK-26642][connector/pulsar] Fix support for non-partitioned topic

### PR DESCRIPTION
**Pull request by golden-yang**
This PR tries to fix the[FLINK-26642](https://issues.apache.org/jira/browse/FLINK-26642)
Pulsar Sink should support partitioned and non-partitioned topics.

### CODE REVIEW

**Comment by syhily** [comment]
Tks for your fix. LGTM

File: flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicMetadataListener.java
Comment (syhily) [inline flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicMetadataListener.java:125]: `for (int i = 0; i < partitionNums; i++) {`

Comment (golden-yang) [reply flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicMetadataListener.java]: Thanks, I forgot about this variable. this has been fixed

File: flink-connectors/flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicMetadataListenerTest.java
Comment (golden-yang) [reply flink-connectors/flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicMetadataListenerTest.java]: OK, I changed the method to Collections.

**Review by fapaul — APPROVED** [review]
Thanks for the fix. I would like to wait for a green CI before merging but I think it makes sense to also backport this fix to the release-1.15 branch. @golden-yang Can you create an PR also for the release-1.15 branch?

**Comment by golden-yang** [comment]
@flinkbot run azure

**Comment by golden-yang** [comment]
> 
CI report successed.  Another PR for release-1.15 is over here [pull-19168](https://github.com/apache/flink/pull/19168)
@fapaul PTAL

**Pull request merged**
